build-support/javascriptstyle/README.md

```diff
@@ -0,0 +1,62 @@
+# Setting up JavaScript Style Checker
+
+1. Download the JavaScriptStyle package from https://github.com/pantsbuild/binaries/build-support/scripts/javascriptstyle
+2. Unpack the javascriptstyle.tgz into your pants support directory
+3. Configure pants.ini
+4. Add new rules
+5. Add plugins to extend beyond the default ruleset
+6. Blacklisting
+
+## Configuring pants.ini
+
+	[lint.javascriptstyle]
+	javascriptstyle_dir: %(pants_supportdir)s/javascriptstyle
+	skip: False
+	fail_slow: False
+
+	[fmt.javascriptstyle]
+	javascriptstyle_dir: %(pants_supportdir)s/javascriptstyle
+	skip: False
+	fail_slow: Falsertdir)s/javascriptstyle
+
+
+## Adding new rules
+
+Add new rules to the .eslintrc file under the "rules" section.
+A full list of eslint rules can be found at http://eslint.org/docs/rules
+
+For plugins, you will need to append the plugin name followed by a '/'.
+
+    "react/jsx-indent": [2, 2]
+
+
+## Adding new plugins
+
+New plugins will need to be installed in the package.json and added to the .eslintrc
+
+To add react plugin to package.json:
+
+    yarn add eslint-plugin-react
```

Comment:
The fact that people will be mutating the `javascriptstyle` package after downloading it is pretty unfortunate, as it seems pretty likely that each user will end up with a tiny fork of the project.

Would it be possible to do this for them automatically? Basically, by downloading the `javascriptstyle` package for them using `binutils` (similar to how we download the `cloc.pl` script: https://github.com/pantsbuild/pants/blob/681c6a4bdbb6e491c8ebad96cf27dd0507f1a572/src/python/pants/backend/graph_info/tasks/cloc.py#L36-L38), and then unpacking it into the task's workdir, and installing any plugins specified in `pants.ini`?

Reply:
We can install the javascriptstyle package for them automatically, but we will lose deterministic installation if we specify additional plugins through pants.ini. Having a single mutable package also gives us version control on dependencies through the yarn.lock file.

Reply:
Hm... yea, that's a pretty critical property. But would calling `yarn add eslint-plugin-react==1.2.3` (ie: with a precise version) result in a stable set of deps for that version? Or is the lock file a local only concept.

Reply:
Yes, the lock file is a local construct. The lock file will get updated after `yarn add eslint-plugin-react==1.2.3`. The typical usage is to make some changes to a package and have yarn automatically generate a lock file as a byproduct of the change. Check in lock file into repository and let other consumers use the lock file to reproduce the same installation. In short, `yarn add` is not guaranteed to generate the same lock file even when specifying a version number or even when pinned to a specific resolved path.

Reply:
Blargh.

Reply:
Alright: please address my other comments and fix the merge conflict, and then I think we can get this landed.

Reply:
Done